heap/heap.py

```diff
@@ -1,21 +1,44 @@
 class Heap:
-  def __init__(self):
-    self.storage = []
+    def __init__(self):
+        self.storage = []
 
-  def insert(self, value):
-    pass
+    def insert(self, value):
+        self.storage.append(value)
+        self._bubble_up(self.get_size() - 1)
 
-  def delete(self):
-    pass
+    def delete(self):
+        if self.get_size() == 1:
+            return self.storage.pop()
+        temp = self.storage[0]
+        self.storage[0] = self.storage.pop()
+        self._sift_down(0)
+        return temp
```

Comment:
I wonder if this could be shorter. Like, you don't really use self.storage[0] for anything else so maybe line 12 could be temp = self.storage.pop(). Also, it looks like you're returning self.storage.pop() regardless of self.get_size()...but then again, it is easy to see your thought process this way. Either way, well done, Jun!

Reply:
temp is storing the original max, where as line 13 is setting the last leaf node as the 0 index, it's a completely different process! I'm not too sure what you meant by the returning self.storage.pop() returning, that only happens if the size is 1, and a sift_down would not be necessary in that case